e2e/tests/uniswap_trade_test.rs

```diff
@@ -59,7 +59,13 @@ async fn test_with_ganache() {
     let uniswap_router = UniswapV2Router02::deployed(&web3)
         .await
         .expect("Failed to load deployed UniswapFactory");
-    let gp_settlement = GPv2Settlement::deployed(&web3)
+
+    // I took this private key from the gp-v1 e2e test. Not sure why this actually works.
+    // Does Ganache guarantee that it works?
+    let private_key = "4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d"
```

Comment:
If ganache is started with the default seed phrase, this will be the 0th account in the list. I don't think we should rely on this though. The more general way to use accounts is via the `Accounts` type. On ganache accounts are "unlocked" meaning they can sign transactions. For xDAI/mainnet/Rinkeby we can create an offline account that is initialized with a private key.

I think `get_settlement_contract` should take `Account` as an argument instead of private key, so that we can pass account[0] in the e2e test.